Fix audio/video not disposing when going to a different screen

tneotia opened the pull request:
fixes #610, fixes #752, fixes #908

I can't reproduce the issue so it needs testing.

Edit: According to @mvn-hainguyen-dn this solution works.

yucomds commented:
Hi @tneotia, when do you think this PR will be merged in the master branch?

tneotia commented:
@yucomds I don't know... @erickok reviews the PRs but he's a bit busy right now so when he's available then it will be reviewed and merged.

If you really need this then you can always use a git dependency pointing to my fork of flutter_html with this branch.

erickok commented:
I think we should clean up any lingering listeners ourselves, one we go the statefull widget way. That should be sufficient, no?

erickok commented:
We still need to fix this but I was hoping to do that after modularization.

tneotia commented:
@erickok I just overhauled this PR a bit:

1. `Html` is now a `StatefulWidget` so we can dispose everything internally
2. I added a `disposeAll` method in case someone would need to dispose all controllers being used by all `Html` widgets
3. I added convenience getters for users to get controllers being used by `Html` widgets so things like #862 can be done by users.

erickok commented:
@tneotia This is probably the first thing we need to merge now. Can you fix the conflict? than I will approve and merge everything.

tneotia commented:
@erickok done :)

tneotia commented:
@erickok done.
